### PR TITLE
Add additional context icon

### PR DESCRIPTION
**Pull request by bryancunningham-okta**
[OKTA-681936](https://oktainc.atlassian.net/browse/OKTA-681936)

## Summary
- Add component to render an `i` icon and tooltip for additional context

## Testing & Screenshots
image

- [x] I have confirmed this change with my designer and the Odyssey Design Team.

### CODE REVIEW

File: packages/odyssey-storybook/src/components/odyssey-mui/AdditionalContextIcon/AdditionalContextIcon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      const icon = canvas.getByTitle(
+        "Use the `i` icon to give details that might not be critical but are 'good to know'"
+      );
+      userEvent.hover(icon);
+      await axeRun(actionName);
+      userEvent.unhover(icon);
+    });
+  };
+
+const Template: StoryObj<AdditionalContextIconProps> = {
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/AdditionalContextIcon/AdditionalContextIcon.stories.tsx:77]: I believe the generic typing for al story objects should be `StoryObj<typeof AdditionalContextIcon>`. I recall the typings like the one you are currently using to type this generic were to get around some storybook bug, but the suggestion here should work without the need for `AdditionalContextIconProps` here and other places.

**Review by chrispulsinelli-okta — COMMENTED** [review]
Curious what your thoughts are regarding the naming of the component I left and provided a couple of suggestions for fixes.

Will take a look again once you reply. 
Nice work!

File: packages/odyssey-react-mui/src/AdditionalContextIcon.tsx
[the remark this answers is not in the document]
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/AdditionalContextIcon.tsx:1]: @chrispulsinelli-okta Yeah, I agree. What do you think about something like `InfoIcon` or something like that?

File: packages/odyssey-react-mui/src/AdditionalContextIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+const IconContainer = styled("span", {
+  shouldForwardProp: (prop) =>
+    prop !== "type" && prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  display: inline-flex;
+  cursor: pointer;
+  border: 1px solid transparent;
```
[the remark this answers is not in the document]
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/AdditionalContextIcon.tsx:45]: @chrispulsinelli-okta for a transparent border? No, we do not

**Review by jordankoschei-okta — DISMISSED** [review]

File: packages/odyssey-react-mui/src/AdditionalContextIcon.tsx
[the remark this answers is not in the document]
Comment (chrispulsinelli-okta) [reply packages/odyssey-react-mui/src/AdditionalContextIcon.tsx:1]: Works for me

File: packages/odyssey-react-mui/src/IconWithTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    outline: ${({ odysseyDesignTokens }) =>
+      `${odysseyDesignTokens.FocusOutlineWidthMain} ${odysseyDesignTokens.FocusOutlineStyle} transparent`};
+    outline-offset: ${({ odysseyDesignTokens }) =>
+      odysseyDesignTokens.FocusOutlineOffsetTight};
+  }
+
+  svg {
+    display: flex;
+  }
+`;
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-react-mui/src/IconWithTooltip.tsx:52]: This, for some reason, causes issues in some consumers of this package. Can you refactor this to be like:
```
const IconContainer = styled("span", {
shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
  })<{
    odysseyDesignTokens: DesignTokens;
  }>(({ odysseyDesignTokens }) => ({
    borderColor: odysseyDesignTokens.FocusOutlineColorPrimary,
    boxShadow: `0 0 0 1px ${odysseyDesignTokens.FocusOutlineColorPrimary}`,
    // ...rest of them
}));
```

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+const storybookMeta: Meta<IconWithTooltipProps> = {
+  title: "MUI Components/IconWIthTooltip",
+  component: IconWithTooltip,
+  argTypes: {
+    IconComponent: {
+      description: "The icon to render",
+      table: {
+        type: {
+          summary: "ReactNode",
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:30]: It's already a `ReactNode` here 😆

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    testId: "tooltip",
+  },
+  decorators: [MuiThemeDecorator],
+  tags: ["autodocs"],
+};
+
+export default storybookMeta;
+
+const showTooltip =
+  ({ canvasElement, step }: PlaywrightProps<IconWithTooltipProps>) =>
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:78]: Should this be `PlaywrightProps<typeof IconWithTooltip>`?

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    });
+  };
+
+export const Default: StoryObj<typeof IconWithTooltip> = {
+  args: {
+    tooltipText:
+      "Use the `i` icon to give details that might not be critical but are 'good to know'",
+  },
+  play: async ({ canvasElement, step }) => {
+    showTooltip({ canvasElement, step })("IconWithTooltip");
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:97]: `await`?

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+const showTooltip =
+  ({ canvasElement, step }: PlaywrightProps<IconWithTooltipProps>) =>
+  async (actionName: string) => {
+    await step("show the tooltip on hover", async () => {
+      const canvas = within(canvasElement);
+      const icon = canvas.getByTitle(
+        "Use the `i` icon to give details that might not be critical but are 'good to know'",
+      );
+      userEvent.hover(icon);
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:85]: Can you check if `hover` is async? As well as `unhover`? May need to await them

**Review by chrispulsinelli-okta — COMMENTED** [review]
Overall LGTM. There's a couple of things to update but otherwise happy to approve once you've got the suggested changes in.

File: packages/odyssey-react-mui/src/IconWithTooltip.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+export type IconWithTooltipProps = {
+  /**
+   * The icon to render. Defaults to `InformationCircleIcon`
+   */
+  IconComponent?: ReactElement;
```
[the remark this answers is not in the document]
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/IconWithTooltip.tsx:58]: I don't ReactNode here I don't think though. I want to make sure this is an actual element. Like, an SVG

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx (hunk 4 above)
Comment (bryancunningham-okta) [reply packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:30]: Ah, typo maybe. Or copy/paste error.

File: packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx (hunk 5 above)
Comment (bryancunningham-okta) [reply packages/odyssey-storybook/src/components/odyssey-mui/IconWithTooltip/IconWithTooltip.stories.tsx:78]: We seemingly just pass the props as the generic here on other cases

File: packages/odyssey-react-mui/src/IconWithTooltip.tsx (hunk 3 above)
Comment (chrispulsinelli-okta) [reply packages/odyssey-react-mui/src/IconWithTooltip.tsx:52]: I can't remember specifically, but they appear when running unit tests with the way it's currently declared

**Review by chrispulsinelli-okta — COMMENTED** [review]
The styled component still needs to be updated for it to work correctly in unit tests, from what I have found. I believe the IconComponent type needs to be updated too for reasons I mentioned in a reply.

**Review by chrispulsinelli-okta — APPROVED** [review]

**Pull request merged**